pkg/skaffold/deploy/kpt/kpt.go

```diff
@@ -56,7 +57,14 @@ const (
 
 	kustomizeDownloadLink  = "https://kubernetes-sigs.github.io/kustomize/installation/"
 	kustomizeMinVersion    = "v3.2.3"
-	kustomizeVersionRegexP = `{Version:(\S+) GitCommit:\S+ BuildDate:\d{4}-\d{2}-\d{2}T\d\d:\d\d:\d\dZ GoOs:\S+ GoArch:\S+}`
+	kustomizeVersionRegexP = `{Version:(\S+) GitCommit:\S+ BuildDate:\S+ GoOs:\S+ GoArch:\S+}`
+)
+
+var (
+	kustomizeFurtherGuidance = fmt.Sprintf("Please make sure your local "+
+		"kustomize version >= the official version %v, otherwise some features may not be "+
+		"well supported. You can download the official version "+
+		"from %v", kustomizeMinVersion, kustomizeDownloadLink)
```

Comment:
I don't think it makes sense to have this text as a var since the arguments as positional instead of named.  I think you're assigning this as a var to be used from the tests.

@tejal29 @PriyaModali here's a possible contribution to build failures that we should report upwards in our metrics: incompatible dependency versions.  We have this for kubectl, helm, and now kpt and kustomize.

Reply:
I don't have strong opinions on the var usage. Yeah, this is just to save some lines in the unitests.

Reply:
Updated.

Reply:
For the incompatible dependency versions, I guess it is fine to ignore the conflict if users use kustomize in kpt deployer (like this PR only warns users but doesn't have hard requirements on actions) . This is because we will replace the kustomize here to some kpt builtin functions, so this version check may likely be changed someday in Q1 2021.

Reply:
Ack @briandealwis . 
This is something i discussed with @tstromberg internally. Skaffold needs to provide a framework or mechanism to error out with an error code instead of playing this catch up game for new error messages.
Minikube has this implemented [here](https://github.com/kubernetes/minikube/tree/c459fea74df093f916e644fac55646755ea7f4ec/pkg/minikube/exit)

I am working on this idea. Happy to hear any thoughts.